### DISCUSSION TITLE
Make uploading more reliable.

### DISCUSSION THREAD

**Discussion by jxqqk** [gopro/labs#199] [Q&A]
I am preparing a daily photo timelapse in raw with a HERO11, following the discussions here I have found a very reliable way for that, an external timer for the power on and off with a very simple boot script (just !SQ), all the settings on camera.
It would be nice to upload the pictures on the cloud, for that reason I now tried a little more advanced boot script (!SQ!timeE!U).
The script works but the transfer always fail, sometimes do not start at all, sometimes after few seconds stops; it also fails in the same way selecting "manual upload" on camera or sending a !U by QRcode.
If I shot a jpeg photo timelapse it works but never with a raw photo timelapse.
There is something I can do to make the upload more reliable ad usable?
Thank you.
JM

**Comment by dnewman-gpsw** [comment]
If it is failing with manual upload, that suggests it is something more core, not Labs related. I'm not aware if .GPR work with uploads. Does the script work with JPGs only? There is definitely a bug in repeating scripts (!R) that also upload (fix in a future Labs), but the boot technique you are using will bypass that error.

**Reply by jxqqk** [reply]
I agree, it looks a core problem, I will contact support for that.
If this is a raw timelapse upload limitation, assuming !timeSQ works, it is possible to send a command like !600SQ!600SQ!600SQ!600SQ...!U to capture single photos (not timelapse)?
How long the command can be?
Thank you.
JM

**Reply by dnewman-gpsw** [reply]
There are issues with waking from Labs initiated sleeps with the !U working reliable on HERO11 (so it not the repeat, but sleeps.)  It has been fixed in a pending version.  

Commands can be up 128 characters, from a QR Code, effectively longer if you linking script together.

**Reply by jxqqk** [reply]
Thank you.
I am testing also the sunrise-sunset feature.
I provide the LLTZ metadata but the command call the gps and since I am indoor the camera waits forever.
Am I doing something wrong?
LM
edit: the timezone format was wrong, not +1.0 but only 1.0, works now.

**Comment by jxqqk** [comment]
Sorry I keep writing here...
I am testing with different script in different time of the day, with linked scripts.
command >11:30<12:30!Lcarica~!Ltimelapse in a boot script
is it the right syntax?
JM

**Reply by dnewman-gpsw** [reply]
That is a good idea, and seems very logical, unfortunately I just confirmed it doesn't work. Seems the else '~' is not compatible with !Lscript. I will look to see if that can be fixed. However there is a work around.  >11:30<12:30!Lcarica!Ltimelapse <- this works.  As the !L command branches away, you don't need the else.

```
if(time > 11:30 && time < 12:30)
   goto carica
else
   goto timelapse
```

Is the same as  

```
if(time > 11:30 && time < 12:30)
   goto carica
goto timelapse
```

**Reply by jxqqk** [reply]
Works without '~' thanks a lot. With different scripts and powering on and off the camera in different moments of the day I try to make the camera shot some pictures in the sunrise, upload the new pictures at noon, shot some pictures in the sunset.
Let see if it works in the next days.
JM

**Comment by jxqqk** [comment]
Sorry another question, is there a command to pause the camera for some time without sleep? After wakeup the boot script start again and this is a problem.
Thank you
JM

**Reply by dnewman-gpsw** [reply]
Not really, the longest pause without sleep is !9N (9 seconds.)  What are you trying to do with this pause?

**Reply by jxqqk** [reply]
Maybe I'm just missing out... Basically I am trying to find a way to have some delay in a boot script, because a boot script "!10SQ" is simply an endless loop of rebooting...
The only solution to have some sort of delay in a boot script is avoid sleeps?
Thank you.
JM

**Reply by dnewman-gpsw** [reply]
!10SQ!R should boot continuously, as the camera has around a 7s boot time for Labs service.  You haven't explained the big picture of why you need a delay that is using power.  If you what a ten second interval timelapse, use a timelapse mode.

**Reply by jxqqk** [reply]
You are right timelapse works, but because cloud service do not support raw timelapse but only jpgs, I am trying to send normal raw photo and not timelapse, also I can decide a different time interval between 5 and 30 min.
I need to take just a few raw pictures near sunrise and near sunset every day with a 15 min delay and upload the raws to the cloud.
I have only a basic external timer.
I am thinking if the sunrise and sunset time can be used in a conditional script.
JM
